ConsoleApp1/Program.cs

```diff
@@ -1,9 +1,361 @@
+using System.Text.RegularExpressions;
+using static ConsoleApp1.Program;
+
+namespace ConsoleApp1
 {
-	internal class Program
-	{
-		static void Main(string[] args)
-		{
-		}
-	}
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            while (true)
+            {
+                DisplayMenu();
+            }
+        }
+
+        static void DisplayMenu()
+        {
+            var path = Path.Combine(Directory.GetCurrentDirectory(), "phonebook.csv");
+
+
+            Console.WriteLine("\tThis is phonebook app");
+
+            Console.WriteLine("\t1. Display phone book records");
+            Console.WriteLine("\t2. Add new record");
+            Console.WriteLine("\t3. Search records");
+            Console.WriteLine("\t4. Edit a record");
+            Console.WriteLine("\t5. Delete a record");
+            Console.WriteLine("\t6. Sort records by anything");
+            Console.WriteLine("\t7. Binary search records by surname");
+            Console.WriteLine("\t0. Exit app");
+
+            var pressedKey = Console.ReadKey();
+            try
+            {
+                switch (pressedKey.Key)
+                {
+                    case ConsoleKey.D1:
+                        var records = ReadPhoneBookRecords(path);
+
+                        DisplayPhoneBook(records);
+
+                        break;
+                    case ConsoleKey.D2:
+                        AddNewRecord(path);
+                        break;
+                    case ConsoleKey.D3:
+                        SearchRecords(path);
+                        break;
+                    case ConsoleKey.D4:
+                        EditRecord(path);
+                        break;
+                    case ConsoleKey.D5:
+                        DeleteRecord(path);
+                        break;
+                    case ConsoleKey.D6:
+                        Sort(path);
+                        break;
+                    case ConsoleKey.D7:
+                        // binary search
+                        int id;
+                        string message = BinarySearch(path, out id) ? $"Surname is in phonebook! Id: {id + 1}" : "Surname was not found";
+                        Console.WriteLine(message);
+                        break;
+                    case ConsoleKey.D0:
+                    default:
+                        Environment.Exit(0);
+                        break;
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Program error occured: {ex.GetType()} {Environment.NewLine} {ex.Message} ");
+            }
+            Console.WriteLine("Press any key to continue");
+            Console.ReadKey();
+            Console.Clear();
+        }
+
+        static (string, string, string)[] ReadPhoneBookRecords(string path)
+        {
+            Console.WriteLine();
+            string[] records;
+            try
```

Comment:
this try makes no sense

Declare exception types and handle them here

Reply:
changed to a more specified one.